exporter/googleservicecontrolexporter/logs.go

```diff
@@ -0,0 +1,493 @@
+		var logEntrySourceLocation scpb.LogEntrySourceLocation
+		err := unmarshalAttribute(sourceLocation, &logEntrySourceLocation)
+		if err != nil {
+			return nil, &attributeProcessingError{Key: SourceLocationAttributeKey, Err: err}
+		}
+		entry.SourceLocation = &logEntrySourceLocation
+		delete(attrsMap, SourceLocationAttributeKey)
+	}
+
+	// parse HttpRequest
+	if httpRequestAttr, ok := attrsMap[HTTPRequestAttributeKey]; ok {
+		httpRequest, err := l.parseHTTPRequest(httpRequestAttr)
+		if err != nil {
+			l.logger.Debug("Unable to parse httpRequest", zap.Error(err))
+		}
+		entry.HttpRequest = httpRequest
+		delete(attrsMap, HTTPRequestAttributeKey)
+	}
+
+	// parse Severity
+	if logRecord.SeverityNumber() < 0 || int(logRecord.SeverityNumber()) > len(severityMapping)-1 {
+		return nil, fmt.Errorf("unknown SeverityNumber %v", logRecord.SeverityNumber())
+	}
+	severityNumber := logRecord.SeverityNumber()
+	// Log severity levels are based on numerical values defined by Otel/GCP, which are informally mapped to generic text values such as "ALERT", "Debug", etc.
+	// In some cases, a SeverityText value can be automatically mapped to a matching SeverityNumber.
+	// If not (for example, when directly setting the SeverityText on a Log entry with the Transform processor), then the
+	// SeverityText might be something like "ALERT" while the SeverityNumber is still "0".
+	// In this case, we will attempt to map the text ourselves to one of the defined Otel SeverityNumbers.
+	// We do this by checking that the SeverityText is NOT "default" (ie, it exists in our map) and that the SeverityNumber IS "0".
+	// (This also excludes other unknown/custom severity text values, which may have user-defined mappings in the collector)
+	if severityForText, ok := otelSeverityForText[strings.ToLower(logRecord.SeverityText())]; ok && severityNumber == 0 {
+		severityNumber = severityForText
+	}
+	entry.Severity = severityMapping[severityNumber]
+
+	// parse remaining OTel attributes to GCP labels
+	for k, v := range attrsMap {
+		if k == LogNameAttributeKey {
+			continue
+		}
+		if _, ok := entry.Labels[k]; !ok {
+			entry.Labels[k] = v.AsString()
+		}
+	}
+
+	// Handle map and bytes as JSON-structured logs if they are successfully converted.
+	switch logRecord.Body().Type() {
+	case pcommon.ValueTypeMap:
+		s, err := structpb.NewStruct(logRecord.Body().Map().AsRaw())
+		if err == nil {
+			entry.Payload = &scpb.LogEntry_StructPayload{StructPayload: s}
+			return entry, nil
+		}
+		l.logger.Warn(fmt.Sprintf("map body cannot be converted to a json payload, exporting as raw string: %+v", err))
+	case pcommon.ValueTypeBytes:
+		s, err := toProtoStruct(logRecord.Body().Bytes().AsRaw())
+		if err == nil {
+			entry.Payload = &scpb.LogEntry_StructPayload{StructPayload: s}
+			return entry, nil
+		}
+		l.logger.Debug(fmt.Sprintf("bytes body cannot be converted to a json payload, exporting as base64 string: %+v", err))
```

Comment:
[nit] The log above is a warn, and this is a debug. We should make them both the same.

Reply:
Done!